### PR TITLE
Task edit and deletion

### PR DESCRIPTION
**Pull request by BobCabossa**

### CODE REVIEW

File: Mobile/lib/Components/TaskEdit.dart [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+                  ],
+                ),
+                SizedBox(height: 16),
+                TextFormField(
+                  enabled: recurring,
+                  decoration: InputDecoration(labelText: 'Recurring Interval'),
+                  onChanged: (value) => recurringInterval = value,
+                  validator: (value) {
+                    if (value == null) {
+                      return 'Please select an end date and time';
```
Comment (krc-coding) [inline Mobile/lib/Components/TaskEdit.dart]: Skal denne tekst ikke opdateres da det er et tal man skal skrive og ikke en dato?

File: Mobile/lib/Components/TaskEdit.dart [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+                  autovalidateMode: AutovalidateMode.onUnfocus,
+                  decoration: InputDecoration(labelText: 'Reward'),
+                  onChanged: (value) => reward = value,
+                  validator: (value) {
+                    if (value == null || value.isEmpty) {
+                      return 'Field is required and cannot be empty';
+                    }
+      
+                    if (double.tryParse(value) == null) {
+                      return 'Please enter a hold number';
```
Comment (krc-coding) [inline Mobile/lib/Components/TaskEdit.dart]: It should be either `whole number` or `integer`
```suggestion
                      return 'Please enter a whole number';
```

File: Mobile/lib/Components/TaskEdit.dart [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+    );
+  }
+
+  void _deleteTask() async {
+    int statusCode = await _appState.deleteTask(oldTask.id);
+    if (statusCode == 204) {
+      Navigator.of(context).pop();
+      Navigator.of(context).pop();
+      Navigator.of(context).pop();
+      _appState.switchPage(_appState.page);
```
Comment (krc-coding) [inline Mobile/lib/Components/TaskEdit.dart]: This line seems redundant, as you're asking it to switch to current page.
```suggestion
```

**Review by krc-coding — CHANGES_REQUESTED** [review]
I haven't checked the code locally yet, but found a few minor things that could be changed.

File: Mobile/lib/Components/TaskEdit.dart (hunk 1 above)
Comment (BobCabossa) [reply Mobile/lib/Components/TaskEdit.dart]: jo da

File: Mobile/lib/Components/TaskEdit.dart (hunk 2 above)
Comment (BobCabossa) [reply Mobile/lib/Components/TaskEdit.dart]: nice i didn't know how to spell that 😄

File: Mobile/lib/Components/TaskEdit.dart (hunk 3 above)
Comment (BobCabossa) [reply Mobile/lib/Components/TaskEdit.dart]: tried to reload the side... didn't work and forgot to remove

**Review by krc-coding — APPROVED** [review]

**Pull request merged**